test/e2e/route_advertisements.go

```diff
@@ -125,17 +125,19 @@ var _ = ginkgo.Describe("BGP: Pod to external server when default podNetwork is
 			framework.ExpectNoError(err, "must get bgpnet subnets")
 			framework.Logf("the network cidrs to be imported are v4=%s and v6=%s", externalServerV4CIDR, externalServerV6CIDR)
 			for _, node := range nodes.Items {
-				ipVer := ""
-				bgpRouteCommand := strings.Split(fmt.Sprintf("ip%s route show %s", ipVer, externalServerV4CIDR), " ")
-				framework.Logf("Checking for server's route in node %s", node.Name)
-				gomega.Eventually(func() bool {
-					routes, err := infraprovider.Get().ExecK8NodeCommand(node.GetName(), bgpRouteCommand)
-					framework.ExpectNoError(err, "failed to get BGP routes from node")
-					framework.Logf("Routes in node %s", routes)
-					return strings.Contains(routes, frrContainerIPv4)
-				}, 30*time.Second).Should(gomega.BeTrue())
-				if isDualStackCluster(nodes) {
-					ipVer = " -6"
+				if isIPv4Supported(f.ClientSet) {
+					ipVer := ""
+					bgpRouteCommand := strings.Split(fmt.Sprintf("ip%s route show %s", ipVer, externalServerV4CIDR), " ")
+					framework.Logf("Checking for server's route in node %s", node.Name)
+					gomega.Eventually(func() bool {
+						routes, err := infraprovider.Get().ExecK8NodeCommand(node.GetName(), bgpRouteCommand)
+						framework.ExpectNoError(err, "failed to get BGP routes from node")
+						framework.Logf("Routes in node %s", routes)
+						return strings.Contains(routes, frrContainerIPv4)
+					}, 30*time.Second).Should(gomega.BeTrue())
+				}
+				if isIPv6Supported(f.ClientSet) || isDualStackCluster(nodes) {
```

Comment:
What is the purpose of `|| isDualStackCluster(nodes)`?
Isn't `if isIPv6Supported(f.ClientSet)` enough on its own?